web/src/app/users/UserEditDialog.tsx

```diff
@@ -19,40 +19,116 @@ interface UserEditDialogProps {
 }
 
 function UserEditDialog(props: UserEditDialogProps): JSX.Element {
-  const { ready: isSessionReady, userID: currentUserID } = useSessionInfo()
+  const defaultValue: Value = {
+    oldPassword: '',
+    newPassword: '',
+    confirmNewPassword: '',
+    isAdmin: props.role === 'admin',
+  }
+
+  const {
+    ready: isSessionReady,
+    userID: currentUserID,
+    isAdmin,
+  } = useSessionInfo()
 
-  const [adminChecked, setAdminChecked] = React.useState(props.role === 'admin')
+  const [value, setValue] = useState(defaultValue)
+  const [errors, setErrors] = useState<FieldError[]>([])
 
-  const [editUser, editUserStatus] = useMutation(mutation, {
-    onCompleted: props.onClose,
+  const [editUser, editUserStatus] = useMutation(updateUserInput, {
+    variables: {
+      input: {
+        id: props.userID,
+        role:
+          defaultValue.isAdmin !== value.isAdmin
+            ? value.isAdmin
+              ? 'admin'
+              : 'user'
+            : null,
+        oldPassword: value.oldPassword !== '' ? value.oldPassword : null,
+        newPassword: value.newPassword !== '' ? value.newPassword : null,
+      },
+    },
   })
 
-  function handleChange(e: React.ChangeEvent<HTMLInputElement>): void {
-    setAdminChecked(e.target.checked)
+  // Checks if any of the password fields are used. Used to skip any unnecessary updateUserPassword mutation
+  function passwordChanged(): boolean {
+    return !(
+      value.oldPassword === '' &&
+      value.newPassword === '' &&
+      value.confirmNewPassword === ''
+    )
```

Comment:
nit: Can we invert this? from `NOT (... AND ...)` to `... OR ...`
```suggestion
    return Boolean(
      value.oldPassword||value.newPassword||value.confirmNewPassword
    )
```